### PR TITLE
Send app name and version

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: ADAL.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>BuildSystemType</key>
+	<string>Latest</string>
```
[the remark this answers is not in the document]
Comment (oldalton) [reply ADAL.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings]: Fixed :)

File: ADAL/src/ADLogger.m [hunk 2]
```diff
@@ -329,9 +353,14 @@ + (void)logToken:(NSString *)token
...
-    [self adalId];
+    [self adalMetadata];
     
-    [s_adalId setObject:value forKey:key];
+    [s_adalFullMetadata setObject:value forKey:key];
+
+    if (s_adalShortMetadata[key])
+    {
+        [s_adalShortMetadata setObject:value forKey:key];
+    }
```
Comment (jasoncoolmax) [inline ADAL/src/ADLogger.m]: I am still thinking, should broker use this API to set the app name and app version? if we do, we may need an API to reset adalMetadata

Comment (oldalton) [reply ADAL/src/ADLogger.m]: What is broker using now for adal version?

Comment (jasoncoolmax) [reply ADAL/src/ADLogger.m]: Broker relies on ADAL to set the adal version, by calling [`ADAuthenticationRequest acquireToken:]`,  [`ADAuthenticationRequest requestCode:]`and `[ADWebAuthRequest send:].

For app name/version, seems broker requires them  to be context/request specific. I am thinking one possible solution is to 
- have readonly properties for app name/version in `ADAuthentionRequest`
- have an overload [ADLogger adalShortMetadata:(NSDictionary *)overwriteDefaultMetadata]
Any other idea? :)

File: ADAL/src/request/ADAuthenticationRequest+WebRequest.m [hunk 3]
```diff
@@ -117,7 +119,7 @@ - (NSString*)generateQueryStringForRequestType:(NSString*)requestType
                                  OAUTH2_REDIRECT_URI, [[_requestParams redirectUri] adUrlFormEncode],
                                  OAUTH2_STATE, state];
     
-    [startUrl appendFormat:@"&%@", [[ADLogger adalId] adURLFormEncode]];
+    [startUrl appendFormat:@"&%@", [[ADLogger adalMetadata] adURLFormEncode]];
```
Comment (jasoncoolmax) [inline ADAL/src/request/ADAuthenticationRequest+WebRequest.m]: we don't send app name/version to auth endpoint? :)

Comment (oldalton) [reply ADAL/src/request/ADAuthenticationRequest+WebRequest.m]: Yeah, I followed the logic from before to minimize changes...

**Pull request merged**